helm/examples/azure/log-analytics/loganalytics-hpcc-logaccessV2.yaml

```diff
@@ -0,0 +1,91 @@
+# Configures HPCC logAccess to target Azure Log Analytics Workspace
+global:
+  logAccess:
+    name: "Azure LogAnalytics LogAccess"
+    type: "AzureLogAnalyticsCurl"
+    #connection:
+      #All connection attributes are optional.
+      #It is preferable to provide connection values as secret values category 'esp', secret name 'azure_logaccess'
+      # NOTE: secret 'azure_logaccess' must include 'aad-client-secret' and it cannot be provided in configuration
+      #
+      #workspaceID: "XYZ"      #ID of the Azure LogAnalytics workspace to query logs from
+                               # Secret value equivalent: 'ala-workspace-id'
+      #clientID: "DEF"         #ID of Azure Active Directory registered application with api.loganalytics.io access - format: 00000000-0000-0000-0000-000000000000
+                               # Secret value equivalent: 'aad-client-id'
+      #tenantID: "ABC"         #The Azure Active Directory Tenant ID, required for KQL API access
+                               # Secret value equivalent: 'aad-tenant-id'
+    logMaps:
+    - type: "global"
+      storeName: "ContainerLogV2"
+      searchColumn: "LogMessage"
+      timeStampColumn: "hpcc_log_timestamp"
+      columnType: "dynamic"
+      columnMode: "ALL"
+    - type: "workunits"
+      searchColumn: "hpcc_log_jobid"
+      columnMode: "DEFAULT"
+      columnType: "string"
```

Comment:
```
    kql.appendf("\n| extend hpcclogfields = extract_all(@\'^([0-9A-Fa-f]+)\\s+(OPR|USR|PRG|AUD|UNK)\\s+(DIS|ERR|WRN|INF|PRO|MET|UNK)\\s+(\\d{4}-\\d{2}-\\d{2}\\s\\d{2}:\\d{2}:\\d{2}\\.\\d+)\\s+(\\d+)\\s+(\\d+)\\s+(UNK|[A-Z]\\d{8}-\\d{6}(?:-\\d+)?)\\s+\\\"(.*)\\\"$', %s)[0]", sourceCol.str());
```

future: I do wonder if HPCC fields embedded within LogEntry/LogMessage, should not be mapped differently.
We should add a jlog function to extract them in a standard accepted way.
Then perhaps this mapping should explicitly say state that it's a named derived field, from where = LogMessage, and how, the how may be by name "jobid", if our jlog extract function returns a map.

Reply:
Absolutely, I believe the best practice is to report the log entry in json from the source.
1 - I've added JLog json support, and have advocated enabling it by default in containerized mode, perhaps we discuss at the offsite
2 - If json format is not enabled, I believe there's a way to parse log messages pre-injection into ContainerLog table, this requires non-trivial schema adjustment for the target environment, worth considering.

Reply:
>Are you suggesting a jlog level function which produces the appropriate mapping for a particular log column? I think you're on to something very useful but we should discuss in person.

No. I was suggesting a standard method added to jlog that takes a line (which may be a json block) and parses it into a name->value map, such that the logaccess yaml says per [hpcc] field something like:

```
    - type: "workunits"
      hpccField: "jobid"
      columnMode: "DEFAULT"
      columnType: "string"
```